src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3628,51 +3774,29 @@ int journal_file_copy_entry(JournalFile *from, JournalFile *to, Object *o, uint6
         };
         boot_id = &o->entry.boot_id;
 
-        n = journal_file_entry_n_items(o);
+        for (uint64_t i = 0;;) {
+                r = journal_file_entry_item_next(from, o, p, &i, NULL, 0, 0, NULL, NULL, NULL);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                n++;
+        }
```

Comment:
With these changes, journal_file_entry_n_items() becomes O(n) instead of O(1) so it seemed like a good idea to remove it to discourage calculating the size of the entry items unless it's absolutely necessary since it's a more expensive operation now..

Reply:
Ah, the function is dropped by this commit. I did not notice that. Thanks.

So, how heavy is this? What is the typical value of `n`?
Maybe it is better to drop the first loop and replace `items = newa(EntryItemEx, n);` with `GREEDY_REALLOC()`?

Reply:
Typical value is between 15-30 I'd guess. I didn't notice any slowdowns during testing. Also, this function is only invoked by the FlushToVar journald varlink method, which I don't think is on the hot path.